src/org/labkey/test/components/ui/grids/CustomizeGridDialog.java

```diff
@@ -0,0 +1,541 @@
+package org.labkey.test.components.ui.grids;
+
+import org.junit.Assert;
+import org.labkey.test.BootstrapLocators;
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.components.UpdatingComponent;
+import org.labkey.test.components.bootstrap.ModalDialog;
+import org.labkey.test.components.html.Checkbox;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Wraps EntityFieldFilterModal.tsx in UI components.
+ */
+public class CustomizeGridDialog extends ModalDialog
+{
+    private final UpdatingComponent linkedComponent;
+
+    private static final String FIELD_NOT_AVAILABLE = "Field name '%s' is not visible in the 'Available Fields' list.";
+    private static final String FIELD_NOT_IN_GRID = "Field with label '%s' is not visible in the 'Shown in Grid' list.";
+
+    public CustomizeGridDialog(WebDriver driver, UpdatingComponent linkedComponent)
+    {
+        super(new ModalDialogFinder(driver));
+        this.linkedComponent = linkedComponent;
+        waitForReady();
+    }
+
+    @Override
+    protected void waitForReady()
+    {
+        WebDriverWrapper.waitFor(()->
+                {
+                    WebElement spinner = BootstrapLocators.loadingSpinner.findWhenNeeded(this);
+                    return !spinner.isDisplayed() &&
+                            elementCache().contentPanelLocator.findElements(this).size() == 2;
```

Comment:
This `findWhenNeeded` -> `isDisplayed` pattern is pretty common so I recently added an `isDisplayed` method to `Locator`.
```suggestion
                    return !BootstrapLocators.loadingSpinner.isDisplayed(this) &&
                            elementCache().contentPanelLocator.findElements(this).size() == 2;
```

Reply:
Updated.